token/services/identity/deserializer/typed.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package deserializer
+
+import (
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/hash"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/core/common"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/driver"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/identity"
+	"github.com/pkg/errors"
+)
+
+type TypedVerifierDeserializer interface {
+	DeserializeVerifier(typ string, raw []byte) (driver.Verifier, error)
+	Recipients(id driver.Identity, typ string, raw []byte) ([]driver.Identity, error)
+	GetOwnerAuditInfo(id driver.Identity, typ string, raw []byte, p driver.AuditInfoProvider) ([][]byte, error)
+}
+
+// AuditMatcherDeserializer deserializes raw bytes into a matcher, which allows an auditor to match an identity to an enrollment ID
+type AuditMatcherDeserializer interface {
+	GetOwnerMatcher(raw []byte) (driver.Matcher, error)
+}
+
+type TypedVerifierDeserializerMultiplex struct {
+	deserializers            map[string]TypedVerifierDeserializer
```

Comment:
We could use an alias to link this value to the `Type` property and even provide constants for it

Reply:
👍